### PR TITLE
Message improvements

### PR DESCRIPTION
**Pull request by SnazzyPine25**
• Added Message#tts
• Added Message#mentioned_everyone
• Added Message#edited_timestamp
• Added Message#edited
• Added alias for Message#author (writer)
• Added Embeds Support
• Fixed grammar error
• Fixed coding style in Message

### CODE REVIEW

**Comment by SnazzyPine25** [comment]
#205'

File: lib/discordrb/data.rb [hunk 1]
```diff
@@ -1243,6 +1243,99 @@ def update_channel_data
...
+    # @return [String] the description of the embed object.
+    attr_reader :description
+
+    # @return [String] the type of the embed object.
+    attr_reader :type
+
+    # @return [EmbedProvider] the provider of the embed object.
+    attr_reader :provider
+
+    # @return [Attachment] the thumbnail of the embed object. This is limited to `url`, `proxy_url`, `width` and `height`.
```
Comment (meew0) [inline lib/discordrb/data.rb]: This type parameter is wrong, it should be `EmbedThumbnail`

File: lib/discordrb/data.rb [hunk 2]
```diff
@@ -1315,20 +1416,39 @@ class Message
...
-    alias_method :to_s, :content
 
     # @!visibility private
     def initialize(data, bot)
       @bot = bot
       @content = data['content']
       @channel = bot.channel(data['channel_id'].to_i)
       @pinned = data['pinned']
+      @tts = data['tts']
+      @mentioned_everyone = data['mentioned_everyone']
```
Comment (meew0) [inline lib/discordrb/data.rb]: Without having tested it, I believe the name for the field is `mention_everyone`, not `mentioned_everyone`.

Comment (meew0) [inline lib/discordrb/data.rb]: The docs confirm it: https://discordapp.com/developers/docs/resources/channel#message-object

File: lib/discordrb/data.rb [hunk 3]
```diff
@@ -1315,20 +1443,39 @@ class Message
...
+    attr_reader :tts
+    alias_method :tts?, :tts
+
+    # @return [true, false] whether the message was edited or not.
+    attr_reader :edited
+    alias_method :edited?, :edited
+
+    # @return [true, false] whether the message mentioned everyone or not.
+    attr_reader :mention_everyone
+    alias_method :mention_everyone?, :mention_everyone
```
Comment (meew0) [inline lib/discordrb/data.rb]: Are all these aliases really necessary...? I'd prefer two at maximum; this one (`mention_everyone?`) and also `mentions_everyone?` for slightly more grammatical code. All the ones without question marks violate Ruby style in a way, and the `all` thing is just confusing in my opinion.

File: lib/discordrb/data.rb [hunk 4]
```diff
@@ -1243,6 +1243,126 @@ def update_channel_data
...
+    attr_reader :type
+
+    # @return [EmbedProvider, nil] the provider of the embed object. `nil` is there is no provider
+    attr_reader :provider
+
+    # @return [EmbedThumbnail, nil] the thumbnail of the embed object. `nil` is there is no thumbnail
+    attr_reader :thumbnail
+
+    # @return [EmbedAuthor, nil] the author of the embed object. `nil` is there is no author
+    attr_reader :thumbnail
```
Comment (meew0) [inline lib/discordrb/data.rb]: `:author`

File: lib/discordrb/data.rb [hunk 5]
```diff
@@ -1243,6 +1243,126 @@ def update_channel_data
...
+    # @return [Symbol] the type of the embed object.
+    attr_reader :type
+
+    # @return [EmbedProvider, nil] the provider of the embed object. `nil` is there is no provider
+    attr_reader :provider
+
+    # @return [EmbedThumbnail, nil] the thumbnail of the embed object. `nil` is there is no thumbnail
+    attr_reader :thumbnail
+
+    # @return [EmbedAuthor, nil] the author of the embed object. `nil` is there is no author
```
Comment (meew0) [inline lib/discordrb/data.rb]: "if there is no"

File: lib/discordrb/data.rb [hunk 6]
```diff
@@ -1243,6 +1243,126 @@ def update_channel_data
...
+    # @return [String, nil] the description of the embed object. `nil` if there is no description
+    attr_reader :description
+
+    # @return [Symbol] the type of the embed object.
+    attr_reader :type
+
+    # @return [EmbedProvider, nil] the provider of the embed object. `nil` is there is no provider
+    attr_reader :provider
+
+    # @return [EmbedThumbnail, nil] the thumbnail of the embed object. `nil` is there is no thumbnail
```
Comment (meew0) [inline lib/discordrb/data.rb]: "if there is no"

File: lib/discordrb/data.rb [hunk 7]
```diff
@@ -1243,6 +1243,126 @@ def update_channel_data
...
+    # @return [String, nil] the title of the embed object. `nil` if there is no title
+    attr_reader :title
+
+    # @return [String, nil] the description of the embed object. `nil` if there is no description
+    attr_reader :description
+
+    # @return [Symbol] the type of the embed object.
+    attr_reader :type
+
+    # @return [EmbedProvider, nil] the provider of the embed object. `nil` is there is no provider
```
Comment (meew0) [inline lib/discordrb/data.rb]: "if there is no"

File: lib/discordrb/data.rb [hunk 8]
```diff
@@ -1243,6 +1243,126 @@ def update_channel_data
...
+    # @return [String] the URL this embed object is based on.
+    attr_reader :url
+
+    # @return [String, nil] the title of the embed object. `nil` if there is no title
+    attr_reader :title
+
+    # @return [String, nil] the description of the embed object. `nil` if there is no description
+    attr_reader :description
+
+    # @return [Symbol] the type of the embed object.
```
Comment (meew0) [inline lib/discordrb/data.rb]: This should give some example values, as `type` is pretty generic

File: lib/discordrb/data.rb [hunk 9]
```diff
@@ -1243,6 +1243,126 @@ def update_channel_data
...
+
+    # @return [EmbedThumbnail, nil] the thumbnail of the embed object. `nil` is there is no thumbnail
+    attr_reader :thumbnail
+
+    # @return [EmbedAuthor, nil] the author of the embed object. `nil` is there is no author
+    attr_reader :thumbnail
+
+    # @!visibility private
+    def initialize(data, message)
+      @bot = bot
```
Comment (meew0) [inline lib/discordrb/data.rb]: `bot` is nowhere defined

File: lib/discordrb/data.rb [hunk 10]
```diff
@@ -1243,6 +1243,126 @@ def update_channel_data
...
+
+    # @return [Integer] the width of this thumbnail file, in pixels.
+    attr_reader :width
+
+    # @return [Integer] the height of this thumbnail file, in pixels.
+    attr_reader :height
+
+    # @!visibility private
+    def initialize(data, embed)
+      @bot = bot
```
Comment (meew0) [inline lib/discordrb/data.rb]: `bot` is nowhere defined

File: lib/discordrb/data.rb [hunk 11]
```diff
@@ -1243,6 +1243,126 @@ def update_channel_data
...
+
+    # @return [String] the provider's name.
+    attr_reader :name
+
+    # @return [String, nil] the URL of the provider. `nil` is there is no URL
+    attr_reader :url
+
+    # @!visibility private
+    def initialize(data, embed)
+      @bot = bot
```
Comment (meew0) [inline lib/discordrb/data.rb]: `bot` is nowhere defined

File: lib/discordrb/data.rb [hunk 12]
```diff
@@ -1243,6 +1243,126 @@ def update_channel_data
...
+
+    # @return [String] the author's name.
+    attr_reader :name
+
+    # @return [String, nil] the URL of the author's website. `nil` is there is no URL
+    attr_reader :url
+
+    # @!visibility private
+    def initialize(data, embed)
+      @bot = bot
```
Comment (meew0) [inline lib/discordrb/data.rb]: `bot` is nowhere defined

File: lib/discordrb/data.rb [hunk 13]
```diff
@@ -1243,6 +1243,126 @@ def update_channel_data
...
+      @message = message
+
+      @url = data['url']
+      @title = data['title']
+      @type = data['type'].to_sym
+      @description = data['description']
+      @provider = data['provider'].nil? ? nil : EmbedProvider.new(data['provider'], self)
+      @thumbnail = data['thumbnail'].nil? ? nil : EmbedThumbnail.new(data['thumbnail'], self)
+      @thumbnail = data['author'].nil? ? nil : EmbedAuthor.new(data['author'], self)
```
Comment (meew0) [inline lib/discordrb/data.rb]: `@author`

**Pull request merged**